### ISSUE TITLE
Allow plugins to declare a release track

### ISSUE THREAD

**Issue by natm** [netbox-community/netbox#17857]
### NetBox version

v4.1.4

### Feature type

New functionality

### Triage priority

I volunteer to perform this work (if approved)

### Proposed functionality

When defining a plugin it would be helpful if along with the version, the release track could be optionally declared, for example `dev`, `release`, etc.

```python
from netbox.plugins import PluginConfig

class FooBarConfig(PluginConfig):
    name = 'foo_bar'
    verbose_name = 'Foo Bar'
    description = 'An example NetBox plugin'
    version = '0.0.1'
    author = 'Jeremy Stretch'
    author_email = 'author@example.com'
    base_url = 'foo-bar'
    release_track = 'release'

config = FooBarConfig
```

### Use case

When installed plugins are listed the track could be displayed along with the version. This could be used to distinguish if a plugin has been installed from a published release or is running from a local development clone.

### Database changes

None

### External dependencies

None

**Comment by jeremystretch** [comment]
Makes sense to me. Do we want to go as far as declaring a set of valid values for this attribute, or leave it up to the individual plugins?

**Comment by natm** [comment]
> Makes sense to me. Do we want to go as far as declaring a set of valid values for this attribute, or leave it up to the individual plugins?

Yes it would be nice if the options could be constrained to `dev` and `release` with a default of `release` if the field is not specified.
